src/nvim/shada.c

```diff
@@ -3984,13 +3984,15 @@ static ShaDaReadResult shada_read_next_item(ShaDaReadDef *const sd_reader,
       entry->data.buffer_list.buffers =
           xcalloc(unpacked.data.via.array.size,
                   sizeof(*entry->data.buffer_list.buffers));
+      // XXX: The macros (CHECK_KEY_IS_STR, ADDITIONAL_KEY, …) depend on the
+      // variable names `unpacked` and `i`.  So we must save/restore them.
+      msgpack_unpacked save_unpacked = unpacked;
       for (size_t i = 0; i < unpacked.data.via.array.size; i++) {
         entry->data.buffer_list.size++;
-        msgpack_unpacked unpacked_2 = (msgpack_unpacked) {
+        unpacked = (msgpack_unpacked) {
           .data = unpacked.data.via.array.ptr[i],
         };
         {
-          msgpack_unpacked unpacked = unpacked_2;
```

Comment:
This shadowing was intentional, because macros like `ADDITIONAL_KEY` implicitly depend on the variable names `unpacked` and `i`. 

To avoid the shadowing, overwrite `unpacked` then restore it later from `save_unpacked`. (Adjusting the macros was too much churn/risk.)